mmedit/models/video_interpolater/basic_interpolater.py

```diff
@@ -0,0 +1,289 @@
+                'target_path' (str of a path) or 'inputs_path' (list of str)
+            iteration (int): Iteration for the saving image name.
+                Default: None.
+            save_path (str): Path to save image. Default: None.
+            output (Tensor): Output image.
+        """
+
+        if output.ndim == 4:  # an image, key = 000001/0000 (Vimeo-90K)
+            img_name = meta[0]['key'].replace('/', '_')
+            if isinstance(iteration, numbers.Number):
+                save_path = osp.join(save_path,
+                                     f'{img_name}-{iteration + 1:06d}.png')
+            elif iteration is None:
+                save_path = osp.join(save_path, f'{img_name}.png')
+            else:
+                raise ValueError('iteration should be number or None, '
+                                 f'but got {type(iteration)}')
+            mmcv.imwrite(tensor2img(output), save_path)
+        elif output.ndim == 5:  # a sequence, key = 000
+            folder_name = meta[0]['key'].split('/')[0]
+            for i in range(0, output.size(1)):
+                if isinstance(iteration, numbers.Number):
+                    save_path_i = osp.join(save_path, folder_name,
+                                           f'{i:08d}-{iteration + 1:06d}.png')
+                elif iteration is None:
+                    save_path_i = osp.join(save_path, folder_name,
+                                           f'{i:08d}.png')
+                else:
+                    raise ValueError('iteration should be number or None, '
+                                     f'but got {type(iteration)}')
+                mmcv.imwrite(tensor2img(output[:, i, :, :, :]), save_path_i)
+
+    def forward_dummy(self, img):
+        """Used for computing network FLOPs.
+
+        Args:
+            img (Tensor): Input image.
+
+        Returns:
+            Tensor: Output image.
+        """
+        out = self.generator(img)
+        return out
+
+    def train_step(self, data_batch, optimizer):
+        """Train step.
+
+        Args:
+            data_batch (dict): A batch of data.
+            optimizer (obj): Optimizer.
+
+        Returns:
+            dict: Returned output.
+        """
+        outputs = self(**data_batch, test_mode=False)
+        loss, log_vars = self.parse_losses(outputs.pop('losses'))
+
+        # optimize
+        optimizer['generator'].zero_grad()
+        loss.backward()
+        optimizer['generator'].step()
+
+        outputs.update({'log_vars': log_vars})
+        return outputs
+
+    def val_step(self, data_batch, **kwargs):
+        """Validation step.
+
+        Args:
+            data_batch (dict): A batch of data.
+            kwargs (dict): Other arguments for ``val_step``.
+
+        Returns:
+            dict: Returned output.
+        """
+        output = self.forward_test(**data_batch, **kwargs)
+        return output
+
+    @staticmethod
+    def generate_frames(input_images, output_images):
```

Comment:
This function will be used in the demo right?

By the way, how does the demo split a video into multiple inputs?